core/src/main/java/cucumber/runtime/arquillian/ArquillianCucumber.java

```diff
@@ -120,223 +127,108 @@ protected void runChild(final FrameworkMethod method, final RunNotifier notifier
-        if (featuresIs != null) {
-            final BufferedReader reader = new BufferedReader(new InputStreamReader(featuresIs));
-            String line;
-
-            while ((line = reader.readLine()) != null) {
-                line = line.trim();
-                if (line.isEmpty()) {
-                    continue;
-                }
-
-                final Set<Object> resourceFilters = new HashSet<Object>(filters);
-                final PathWithLines pathWithLines = new PathWithLines(line);
-                resourceFilters.addAll(pathWithLines.lines);
-                builder.parse(new ClassLoaderResource(tccl, pathWithLines.path), new ArrayList<Object>(resourceFilters));
-            }
-        } else { // client side
-            for (final Map.Entry<String, Collection<URL>> entry : Features.createFeatureMap(CucumberConfiguration.instance().getTempDir(), cukespaceConfig.getProperty(CucumberConfiguration.FEATURE_HOME), clazz, tccl).entrySet()) {
-                final PathWithLines pathWithLines = new PathWithLines(entry.getKey());
-                final Set<Object> resourceFilters = new HashSet<Object>(filters);
-                resourceFilters.addAll(pathWithLines.lines);
-                for (final URL url : entry.getValue()) {
-                    builder.parse(new URLResource(pathWithLines.path, url), new ArrayList<Object>(resourceFilters));
-                }
-            }
-        }
-
-        if (cucumberFeatures.isEmpty()) {
-            throw new IllegalArgumentException("No feature found");
-        }
-
-        final RuntimeOptions runtimeOptions;
-        if (clazz.getAnnotation(Cucumber.Options.class) != null || clazz.getAnnotation(CucumberOptions.class) != null) { // by class setting
-            final RuntimeOptionsFactory runtimeOptionsFactory = new RuntimeOptionsFactory(clazz, OPTIONS_ANNOTATIONS);
-            runtimeOptions = runtimeOptionsFactory.create();
-            cleanClasspathList(runtimeOptions.getGlue());
-            cleanClasspathList(runtimeOptions.getFeaturePaths());
-        } else if (cukespaceConfig.containsKey(CucumberConfiguration.OPTIONS)) { // arquillian setting
-            runtimeOptions = new RuntimeOptions(new Env("cucumber-jvm"), asList((cukespaceConfig.getProperty(CucumberConfiguration.OPTIONS, "--strict") + " --strict").split(" ")));
-        } else { // default
-            runtimeOptions = new RuntimeOptions(new Env("cucumber-jvm"), asList("--strict", "-f", "pretty", areColorsNotAvailable(cukespaceConfig)));
-        }
-
-        final boolean reported = Boolean.parseBoolean(cukespaceConfig.getProperty(CucumberConfiguration.REPORTABLE, "false"));
+    public void runCucumber(final Object testInstance, final RunNotifier runNotifier) throws Exception {
+        
+    	final Class<?> javaTestClass = getTestClass().getJavaClass();
+        final ClassLoader threadContextClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
ok so maybe just call it loader since this is a weak solution to get the running loader when executing tests (thread context part is not the goal), wdyt?